### PR TITLE
Added scroll to top button for reports page.

### PR DESCRIPTION
**Pull request by JacOng17**
Added scroll to top button for reports page as some are super long.

**Normal screen**
Screenshot from 2019-10-25 12-49-23

**Mobile**
Screenshot from 2019-10-25 12-49-49

### CODE REVIEW

**Comment by Br3nda** [comment]
i think you have the screen shots mixed up

File: assets/js/scroll-to-top-button.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//Get the button:
+mybutton = document.getElementById("myBtn");
```
Comment (Br3nda) [inline assets/js/scroll-to-top-button.js]:
```suggestion
var mybutton = document.getElementById("myBtn");
```

File: _layouts/post.html [hunk 2]
```diff
@@ -29,6 +29,7 @@ <h1 class="post-title" itemprop="name headline">{{ page.title }}</h1>
       {% endif %}
 
       {{ content }}
+      <button onclick="topFunction()" id="myBtn" title="Go to top">Back to Top</button>
```
Comment (Br3nda) [inline _layouts/post.html]: how about a name for `myBtn` that describes what it is. eg. `scrollToTopBtn`

File: assets/js/scroll-to-top-button.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Get the button:
```
Comment (Br3nda) [inline assets/js/scroll-to-top-button.js]:
```suggestion
/* eslint-env browser */

// Get the button:
```

Comment (JacOng17) [reply assets/js/scroll-to-top-button.js]: What does that do? :stuck_out_tongue:

Comment (Br3nda) [reply assets/js/scroll-to-top-button.js]: eslint is the linter.

i think this comment will tell eslint that this javascript runs in a browser, so it is okay to expect `window` and `document` to exist without defining them.

Comment (JacOng17) [reply assets/js/scroll-to-top-button.js]: :smile:

File: assets/js/scroll-to-top-button.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+// Get the button:
+var mybutton = document.getElementById("scrollToTopBtn");
+
+// When the user scrolls down 20px from the top of the document, show the button
+window.onscroll = function() {
+  scrollFunction()
+};
+
+topFunction();
```
Comment (Br3nda) [inline assets/js/scroll-to-top-button.js]: I think this makes the browser call this method everytime this file is loaded. Are you sure you want this?

Comment (JacOng17) [reply assets/js/scroll-to-top-button.js]: Well, I didn't add that in at first...but CodeFactor is complaining that `topFunction()` is not called.

Happy for suggestions as I'm not too sure what to do :frowning_face:

**Comment by Br3nda** [comment]
Code climate thinks its not being called, but it is in the html. Maybe there's a config or some other way to convince eslint. But if there isn't, i dont think we should call it on page load like this.

**Comment by Br3nda** [comment]
Another way is to add the handler to the button using javascript on the js file, instead of using onclick.

File: assets/js/scroll-to-top-button.js (hunk 4 above)
Comment (JacOng17) [inline assets/js/scroll-to-top-button.js]:
```suggestion
```

File: assets/js/scroll-to-top-button.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+/* eslint-env browser */
+
+// Get the button:
+var mybutton = document.getElementById("scrollToTopBtn");
+
+// When the user scrolls down 20px from the top of the document, show the button
+window.onscroll = function() {
+  scrollFunction()
+};
+
```
Comment (JacOng17) [inline assets/js/scroll-to-top-button.js]:
```suggestion
```

**Pull request merged**